src/index.ts

```diff
@@ -199,6 +213,30 @@ let cachedIsAvailableResponse: Promise<boolean> | undefined;
  * Determine if the metadata server is currently available.
  */
 export async function isAvailable() {
+  if (process.env.METADATA_SERVER_DETECTION) {
+    const value =
+      process.env.METADATA_SERVER_DETECTION.trim().toLocaleLowerCase();
+
+    if (!(value in METADATA_SERVER_DETECTION)) {
+      throw new RangeError(
+        `Unknown \`METADATA_SERVER_DETECTION\` env variable. Got \`${value}\`, but it should be \`${Object.keys(
+          METADATA_SERVER_DETECTION
+        ).join('`, `')}\`, or unset`
+      );
+    }
+
+    switch (value as keyof typeof METADATA_SERVER_DETECTION) {
+      case 'assume-present':
+        return true;
+      case 'none':
+        return false;
+      case 'bios-only':
+        return getGCPResidency();
+      case 'ping-only':
```

Comment:
Maybe we discussed but I forgot.. we don't want to have a default where both ping and bios happen sequentially?

Reply:
or default in this case is null and also ping?